File: Solutions/0566/0566.go
func matrixReshape(nums [][]int, r int, c int) [][]int {
	if len(nums) == 0 || r*c != len(nums)*len(nums[0]) {
		return nums
	}

	ans := make([][]int, r)
	for i := range ans {
		ans[i] = make([]int, c)
	}

	k := 0
	for i := 0; i < len(nums); i++ {
		for j := 0; j < len(nums[0]); j++ {
			ans[k/c][k%c] = nums[i][j]
			k++
		}
	}

	return ans

}

File: Solutions/0566/0566.ts
function matrixReshape(nums: number[][], r: number, c: number): number[][] {
    if (nums.length === 0 || r * c !== nums.length * nums[0].length) {
        return nums;
    }

    const ans: number[][] = new Array(r).fill(0).map(() => new Array(c).fill(0));
    let k = 0;

    for (let i = 0; i < nums.length; i++) {
        for (let j = 0; j < nums[0].length; j++) {
            ans[Math.floor(k / c)][k % c] = nums[i][j];
            k++;
        }
    }

    return ans;
};